client/collator/src/lib.rs

```diff
@@ -345,34 +274,37 @@ pub async fn start_collator<Block, RA, BS, Spawner>(
 	RA: ProvideRuntimeApi<Block> + Send + Sync + 'static,
 	RA::Api: CollectCollationInfo<Block>,
 {
-	let collator = Collator::new(
+	let collator_service = CollatorService::new(
 		block_status,
-		Arc::new(spawner),
+		Arc::new(spawner.clone()),
 		announce_block,
 		runtime_api,
+	);
+
+	let collator = Collator::new(
+		collator_service,
 		parachain_consensus,
 	);
 
-	let span = tracing::Span::current();
-	let config = CollationGenerationConfig {
+	let mut request_stream = relay_chain_driven::init(
 		key,
 		para_id,
-		collator: Box::new(move |relay_parent, validation_data| {
-			let collator = collator.clone();
-			collator
-				.produce_candidate(relay_parent, validation_data.clone())
-				.instrument(span.clone())
-				.boxed()
-		}),
-	};
-
-	overseer_handle
-		.send_msg(CollationGenerationMessage::Initialize(config), "StartCollator")
-		.await;
+		overseer_handle,
+	).await;
+
+	let collation_future = Box::pin(async move {
+		let span = tracing::Span::current();
+		while let Some(request) = request_stream.next().await {
+			let collation = collator.clone().produce_candidate(
+				*request.relay_parent(),
+				request.persisted_validation_data().clone(),
+			).instrument(span.clone()).await;
```

Comment:
```suggestion
		while let Some(request) = request_stream.next().await {
			let collation = collator.clone().produce_candidate(
				*request.relay_parent(),
				request.persisted_validation_data().clone(),
			).await;
```

The way you have rewritten the code, makes the manual span instrumentation irrelevant :+1: